[bmalloc] Rename DebugHeap to SystemHeap to reflect production use of system malloc

hyjorc1 opened the pull request:
#### b6f2a98535bdcd4d81c8e6c109fad42d3f8be9b3
<pre>
[bmalloc] Rename DebugHeap to SystemHeap to reflect production use of system malloc
<a href="https://bugs.webkit.org/show_bug.cgi?id=294474">https://bugs.webkit.org/show_bug.cgi?id=294474</a>
<a href="https://rdar.apple.com/153335704">rdar://153335704</a>

Reviewed by Keith Miller.

The bmalloc::DebugHeap was originally intended for debugging, but in practice,
it delegates to the system allocator and is used in production contexts such as
system daemons and long-running processes. To better reflect its actual use,
this patch renames DebugHeap to SystemHeap throughout bmalloc.

Changes include:
- Renamed DebugHeap.{h,cpp} -&gt; SystemHeap.{h,cpp}
- Renamed pas_debug_heap.h -&gt; system_debug_heap.h
- Updated all symbol references: tryGet(), getExisting(), memalign(), realloc(), etc.
- Updated assertions, tests, and libpas glue to match new naming
- Removed libtool warning test for DebugHeap.o from filter-build-webkit_unittest

This is a mechanical rename only. No behavior change.

Canonical link: <a href="https://commits.webkit.org/296271@main">https://commits.webkit.org/296271@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ce48b0ca88ce64b7cc4e40da187d0722273e8e77

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/107932 "24 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/27597 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/18016 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/113143 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/58454 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/109895 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/28292 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/36146 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/81922 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/110880 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/22419 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/97242 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/62353 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/21831 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/15375 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/57896 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/100531 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/91770 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/15441 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/116270 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/106505 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/35004 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/25759 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/90957 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/35380 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/93520 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/90751 "Passed tests") | 
| [  ~~🛠 🧪 merge~~](https://ews-build.webkit.org/#/builders/19/builds/23143 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/35654 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/13407 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/30756 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/17445 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/34902 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/40456 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/130783 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/34645 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/35519 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/38005 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/36306 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

Constellation reviewed (commented):
I think you also need to change `pas_debug_heap` too.

kmiller68 reviewed (approved):
r=me, thanks!

hyjorc1 committed:
[bmalloc] Rename DebugHeap to SystemHeap to reflect production use of system malloc
https://bugs.webkit.org/show_bug.cgi?id=294474
rdar://153335704

Reviewed by Keith Miller.

The bmalloc::DebugHeap was originally intended for debugging, but in practice,
it delegates to the system allocator and is used in production contexts such as
system daemons and long-running processes. To better reflect its actual use,
this patch renames DebugHeap to SystemHeap throughout bmalloc.

Changes include:
- Renamed DebugHeap.{h,cpp} -> SystemHeap.{h,cpp}
- Renamed pas_debug_heap.h -> system_debug_heap.h
- Updated all symbol references: tryGet(), getExisting(), memalign(), realloc(), etc.
- Updated assertions, tests, and libpas glue to match new naming
- Removed libtool warning test for DebugHeap.o from filter-build-webkit_unittest

This is a mechanical rename only. No behavior change.

Canonical link: https://commits.webkit.org/296271@main
